File: solutions/java/code-carl/ch01_array/T2_LC0704_Search.java
/**
 * Practice template for:
 * - LeetCode 704. Binary Search
 *   https://leetcode.cn/problems/binary-search/
 */
public class T2_LC0704_Search {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length - 1, mid;
        while (left <= right) {
            mid = (left + right) / 2;
            if (nums[mid] < target) {
                left = mid + 1;
            } else if (nums[mid] > target) {
                right = mid - 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}

File: templates/code-carl/ch01_array/t2_lc0704_search.py
"""LeetCode 704. Binary Search
https://leetcode.cn/problems/binary-search/
"""

from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        pass


if __name__ == "__main__":
    solution = Solution()
    assert solution.search([-1, 0, 3, 5, 9, 12], 9) == 4
    assert solution.search([-1, 0, 3, 5, 9, 12], 2) == -1
    print("All tests passed.")
